modules/pipeline/pipengine/actionexecutor/manager.go

```diff
@@ -36,32 +45,43 @@ func GetManager() *Manager {
 
 func (m *Manager) Initialize(cfgs chan spec.ActionExecutorConfig) error {
 	m.factory = types.Factory
-	m.executors = make(map[types.Name]types.ActionExecutor)
+	m.executorsByName = make(map[types.Name]types.ActionExecutor)
+	m.kindsByName = make(map[types.Name]types.Kind)
+	m.pools = goroutinepool.New(conf.K8SExecutorPoolSize())
 
 	logrus.Info("pipengine action executor manager Initialize ...")
 
 	for len(cfgs) > 0 {
 		c := <-cfgs
 
-		create, ok := m.factory[types.Kind(c.Kind)]
-		if !ok {
-			return errors.Errorf("unregistered action executor kind: %v", c.Kind)
+		create, err := m.GetCreateFnByKind(types.Kind(c.Kind))
+		if err != nil {
+			return err
+		}
+		kind := types.Kind(c.Kind)
+		if kind.IsK8sKind() {
+			// because clusters are too many, k8s type executor will be created later
+			continue
 		}
 
 		name := types.Name(c.Name)
 		for k, v := range c.Options {
-			logrus.Infof("=> kind [%s], name [%s], option: %s=%s", c.Kind, c.Name, k, v)
+			logrus.Infof("=> kind [%s] option: %s=%s", c.Kind, k, v)
 		}
 
 		actionExecutor, err := create(name, c.Options)
 		if err != nil {
-			logrus.Infof("=> kind [%s], name [%s], created failed, err: %v", c.Kind, c.Name, err)
+			logrus.Infof("=> kind [%s] created failed, err: %v", c.Kind, err)
 			return err
 		}
+		m.Lock()
+		m.kindsByName[name] = kind
+		m.executorsByName[actionExecutor.Name()] = actionExecutor
+		m.Unlock()
 
-		m.executors[name] = actionExecutor
-		logrus.Infof("=> kind [%s], name [%s], created", c.Kind, c.Name)
+		logrus.Infof("=> kind [%s] created", c.Kind)
 	}
+	go m.ListenAndPatchK8sExecutor(context.Background())
```

Comment:
use the context passed in by Initialize.

Reply:
ad